internal/executor/instance/persistentworker/isolation/parallels/vm.go

```diff
@@ -0,0 +1,119 @@
+package parallels
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"strings"
+)
+
+var ErrVMFailed = errors.New("Parallels VM operation failed")
+
+type VM struct {
+	name string
+}
+
+type NetworkAdapterInfo struct {
+	MAC string
+}
+
+type HardwareInfo struct {
+	Net0 NetworkAdapterInfo
+}
+
+type VirtualMachineInfo struct {
+	ID       string
+	Name     string
+	Hardware HardwareInfo
+}
+
+func NewVMClonedFrom(ctx context.Context, vmNameFrom string) (*VM, error) {
+	vm := &VM{
+		name: "cirrus-" + uuid.New().String(),
```

Comment:
Maybe use task id so the error message will be self explanatory and not confusing?

Reply:
It's already there:

https://github.com/cirruslabs/cirrus-cli/blob/ed3e710a768ecb1df15041007a111923f72179f2/internal/worker/task.go#L50

Reply:
But error from `prlctl` are propagated to user and these errors contain this unique VM name without the task id.